From 363accde7fc935fdb84042f9a1ac29ebacf5c1f9 Mon Sep 17 00:00:00 2001
From: Kevin Lin <kevinptt@google.com>
Date: Fri, 6 Mar 2026 22:00:43 +0800
Subject: [PATCH] Revert "spanification: migrate base::ReadUnicodeCharacter
 usage to string_view"

This reverts commit 3cb2fbfc59ddcb2fab82c37d5648eb04cc444e61.
---
 base/strings/utf_string_conversion_utils.h | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/base/strings/utf_string_conversion_utils.h b/base/strings/utf_string_conversion_utils.h
index 2603ef3313..0196778e11 100644
--- a/base/strings/utf_string_conversion_utils.h
+++ b/base/strings/utf_string_conversion_utils.h
@@ -64,6 +64,13 @@ BASE_EXPORT bool ReadUnicodeCharacter(std::string_view src,
                                       size_t* char_index,
                                       base_icu::UChar32* code_point_out);
 
+// TODO(crbug.com/40284755): migrate usage to spanified version.
+// Deprecated. Use span version instead.
+BASE_EXPORT bool ReadUnicodeCharacter(const char* src,
+                                      size_t src_len,
+                                      size_t* char_index,
+                                      base_icu::UChar32* code_point_out);
+
 // Reads a UTF-16 character. The usage is the same as the 8-bit version above.
 BASE_EXPORT bool ReadUnicodeCharacter(std::u16string_view src,
                                       size_t* char_index,
-- 
2.53.0.473.g4a7958ca14-goog

